### DISCUSSION TITLE
How to create interactive shell terminal window in an electron application?

### DISCUSSION THREAD

**Discussion by lanyusan** [atom/atom#22256] [Q&A]
Hi, I have a use case for an electron application that needs to run an interactive shell window inside an Electron application, much like split windows with the left side an application, the right side an interactive shell window.

I need to support linux, windows and also macos. I couldn’t find out how by googling. But for sure it can be done as vs code is able to do it.

Any help much appreciated.

Thanks

**Comment by UziTech** [comment]
You can check out [x-terminal](https://github.com/bus-stop/x-terminal) for an example of how a terminal is added to Atom. Basically use [xterm.js](https://github.com/xtermjs/xterm.js) as the front end display and [node-pty](https://github.com/microsoft/node-pty) as the backend that actually connects to a terminal process.

**Reply by lanyusan** [reply]
Very helpful. Thanks
